src/Cake.Core/IO/ProcessArgumentBuilder.cs

```diff
@@ -83,6 +84,33 @@ public string FilterUnsafe(string source)
         }
 
         /// <summary>
+        /// Copies all the arhuments of the current <see cref="ProcessArgumentBuilder"/> to target <see cref="ProcessArgumentBuilder"/>.
+        /// </summary>
+        /// <param name="target">The copy target.</param>
+        public void CopyTo(ProcessArgumentBuilder target)
```

Comment:
Maybe it makes more sense exposing the tokens as a `IReadOnlyList<IProcessArgument>` and add those two methods as extension methods since they only require read access.